### PR TITLE
IDENTITY-2821 authentication.framework

### PR DESCRIPTION
**Pull request by thariyarox**

### CODE REVIEW

File: components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+            }
+            return;
+        }
+
+        if (tenantDataReceiveURLs != null && tenantDataReceiveURLs.size() > 0) {
+            Tenant[] tenants = new Tenant[0];
+            try {
+                tenants = tenantManager.getAllTenants();
+            } catch (UserStoreException e) {
+                log.error("Retrieving all available tenants failed", e);
```
Comment (prabathabey) [inline components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java]: Is there any specific reason why we don't throw this exception to the upstream logic?

File: components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            }
+            StringBuilder tenantDataReceiverUrlBuilder = new StringBuilder();
+            // Add only the active tenants to the list
+            for (Tenant tenant : tenants) {
+                if (tenant.isActive()) {
+                    tenantDataReceiverUrlBuilder.append(tenant.getDomain() + TENANT_LIST_DATA_SEPARATOR);
```
Comment (prabathabey) [inline components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java]: What if some tenant gets unloaded at some point after this is initialized? How do we handle that sort of an instance?

File: components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java (hunk 1 above)
Comment (johannnallathamby) [inline components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java]: This is not related to tenant loading/unloading. This related to tenant activation/deactivation from admin console UI. In case the tenant was deactivated after initialization the onTenantDeactivation() listener will be called and this list will be updated.

File: components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java (hunk 2 above)
Comment (johannnallathamby) [inline components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java]: This is not related to tenant loading/unloading. This related to tenant activation/deactivation from admin console UI. In case the tenant was deactivated after initialization the onTenantDeactivation() listener will be called and this list will be updated.

File: components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java (hunk 1 above)
Comment (johannnallathamby) [inline components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java]: The idea is that we are gracefully recovering from the server error at this point by continuing with tenant array with length 0.

File: components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java (hunk 2 above)
Comment (prabathabey) [inline components/identity/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/listener/AuthenticationEndpointTenantActivityListener.java]: My bad, I didn't realise this is only applicable for tenant activation/deactivation. Thanks for the explanation.
